File: persephone/provision-service/src/jvm/com/vmware/blockchain/deployment/service/provision/OrchestratorModule.java
/* **************************************************************************
 * Copyright (c) 2019 VMware, Inc.  All rights reserved. VMware Confidential
 * *************************************************************************/
package com.vmware.blockchain.deployment.service.provision;

import java.util.concurrent.ExecutorService;
import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
class OrchestratorModule {

    /**
     * Provide an {@link OrchestratorProvider} instance.
     *
     * @return
     *   a singleton {@link OrchestratorProvider} instance.
     */
    @Provides
    @Singleton
    static OrchestratorProvider providesOrchestratorProvider(ExecutorService executor) {
        return new OrchestratorFactory(executor);
    }
}

File: persephone/provisioning-service/src/jvm/com/vmware/blockchain/deployment/service/provision/OrchestratorModule.kt
/* **************************************************************************
 * Copyright (c) 2019 VMware, Inc. All rights reserved. VMware Confidential
 * **************************************************************************/
package com.vmware.blockchain.deployment.service.provision

import com.vmware.blockchain.deployment.orchestration.OrchestratorProvider
import com.vmware.blockchain.deployment.orchestration.vmware.OrchestratorFactory
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module
class OrchestratorModule {

    /**
     * Provide an [OrchestratorProvider] instance.
     *
     * @return
     *   a singleton [OrchestratorProvider] instance.
     */
    @Provides
    @Singleton
    fun providesOrchestratorProvider(): OrchestratorProvider = OrchestratorFactory
}
